src/Types/ApplePayTypes.res

```diff
@@ -65,68 +67,102 @@ type paymentRequestData = {
+      ~merchantCapabilities=getStrArray(jsonDict, "merchant_capabilities"),
+      ~supportedNetworks=getStrArray(jsonDict, "supported_networks"),
+      ~total=getTotal(jsonDict->getDictFromObj("total")),
+      ~merchantIdentifier=getString(jsonDict, "merchant_identifier", ""),
       (),
     )
   }
 }
 
 let billingContactItemToObjMapper = dict => {
   {
-    addressLines: dict->Utils.getStrArray("addressLines"),
-    administrativeArea: dict->Utils.getString("administrativeArea", ""),
-    countryCode: dict->Utils.getString("countryCode", ""),
-    familyName: dict->Utils.getString("familyName", ""),
-    givenName: dict->Utils.getString("givenName", ""),
-    locality: dict->Utils.getString("locality", ""),
-    postalCode: dict->Utils.getString("postalCode", ""),
+    addressLines: dict->getStrArray("addressLines"),
+    administrativeArea: dict->getString("administrativeArea", ""),
+    countryCode: dict->getString("countryCode", ""),
+    familyName: dict->getString("familyName", ""),
+    givenName: dict->getString("givenName", ""),
+    locality: dict->getString("locality", ""),
+    postalCode: dict->getString("postalCode", ""),
   }
 }
 
 let shippingContactItemToObjMapper = dict => {
   {
-    emailAddress: dict->Utils.getString("emailAddress", ""),
-    phoneNumber: dict->Utils.getString("phoneNumber", ""),
-    addressLines: dict->Utils.getStrArray("addressLines"),
-    administrativeArea: dict->Utils.getString("administrativeArea", ""),
-    countryCode: dict->Utils.getString("countryCode", ""),
-    familyName: dict->Utils.getString("familyName", ""),
-    givenName: dict->Utils.getString("givenName", ""),
-    locality: dict->Utils.getString("locality", ""),
-    postalCode: dict->Utils.getString("postalCode", ""),
+    emailAddress: dict->getString("emailAddress", ""),
+    phoneNumber: dict->getString("phoneNumber", ""),
+    addressLines: dict->getStrArray("addressLines"),
+    administrativeArea: dict->getString("administrativeArea", ""),
+    countryCode: dict->getString("countryCode", ""),
+    familyName: dict->getString("familyName", ""),
+    givenName: dict->getString("givenName", ""),
+    locality: dict->getString("locality", ""),
+    postalCode: dict->getString("postalCode", ""),
   }
 }
+
+let getPaymentRequestFromSession = (~sessionObj, ~componentName) => {
+  let paymentRequest =
+    sessionObj
+    ->Option.flatMap(JSON.Decode.object)
+    ->Option.getOr(Dict.make())
+    ->Dict.get("payment_request_data")
+    ->Option.getOr(Dict.make()->JSON.Encode.object)
+    ->transformKeys(CamelCase)
+
+  let requiredShippingContactFields =
+    paymentRequest
+    ->getDictFromJson
+    ->getStrArray("requiredShippingContactFields")
+
+  if (
+    componentName->getIsExpressCheckoutComponent->not &&
+      requiredShippingContactFields->Array.length !== 0
+  ) {
+    let requiredShippingContactFields =
+      requiredShippingContactFields->Array.filter(item => item !== "postalAddress")
```

Comment:
can you please name these with diff names please .. `requiredShippingContactFields` is used many places